ReactiveCocoa/Swift/Signal.swift

```diff
@@ -680,11 +680,29 @@ public func tryMap<T, U, E>(operation: T -> Result<U, E>)(signal: Signal<T, E>)
 	}
 }
 
+/// Forwards events from `signal` until `interval`. Then if signal isn't completed yet,
+/// errors with `error` on `scheduler`.
+public func timeoutWithError<T, E>(error: E, afterInterval interval: NSTimeInterval, onScheduler scheduler: DateSchedulerType)(signal: Signal<T, E>) -> Signal<T, E> {
+	precondition(interval >= 0)
```

Comment:
Ok. Will commit that soon. 👍
On Sun, Feb 15, 2015 at 2:14 PM Dave Lee <notifications@github.com> wrote:

> In ReactiveCocoa/Swift/Signal.swift
> <https://github.com/ReactiveCocoa/ReactiveCocoa/pull/1756#discussion_r24720644>
> :
>
> > @@ -680,11 +680,29 @@ public func tryMap<T, U, E>(operation: T -> Result<U, E>)(signal: Signal<T, E>)
> >  	}
> >  }
> >
> > +/// Forwards events from `signal` until `interval`. Then if signal isn't completed yet,
> > +/// errors with `error` on `scheduler`.
> > +public func timeoutWithError<T, E>(error: E, afterInterval interval: NSTimeInterval, onScheduler scheduler: DateSchedulerType)(signal: Signal<T, E>) -> Signal<T, E> {
> > +	precondition(interval >= 0)
>
> I like @jspahrsummers <https://github.com/jspahrsummers>'s.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ReactiveCocoa/ReactiveCocoa/pull/1756/files#r24720644>
> .
>